Refactored almost everything

warmwaffles opened the pull request:
There are some major fundemental changes with the library. For instance, now we will be treating the `Wellness::System` similar to how we treat `Rack`.

``` ruby
  system = Wellness::System.new('some_system')
  system.use(Wellness::Services::RedisService, { ... })
  system.use(Wellness::Services::PostgresService, { ... })

  config.middleware.insert_before(
    '::ActiveRecord::QueryCache',
    'Wellness::Middleware',
    system
  )
```

The `Wellness::Services::Base` hasn't changed too much, `#check` is still working just find, `#passed_check` is deprecated, `#failed_check` is deprecated. Currently they will just `#warn` when it is used.

warmwaffles committed:
Refactored almost everything

There are some major fundemental changes with the library. For instance,
now we will be treating the Wellness::System similar to how we treat
Rack.

  system = Wellness::System.new('some_system')
  system.use(Wellness::Services::RedisService, { ... })
  system.use(Wellness::Services::PostgresService, { ... })

  config.middleware.insert_before(
    '::ActiveRecord::QueryCache',
    'Wellness::Middleware',
    system
  )

The Wellness::Services::Base hasn't changed too much, #check is still
working just find, #passed_check is deprecated, #failed_check is
deprecated. Currently they will just #warn when it is used.

warmwaffles committed:
Updated the README

nateklaiber commented on lib/wellness/report.rb:
```diff
@@ -0,0 +1,53 @@
+
+    # @return [TrueClass,FalseClass]
+    def healthy?
+      @services.values.all? { |h| h[:status] == 'HEALTHY' }
```
In the `Services::Base`:https://github.com/warmwaffles/wellness/pull/4/files#diff-7205e553bc54fad6cea882bbc496e6ecR23

Could this method be simplified with:

``` ruby
def healthy?
  @services.all?(&:healthy?)
end
```

This way this class doesn't have to know what _defines_ healthy (`HEALTHY` status), it gets the answer from the service itself.

nateklaiber commented:
I like the changes here. I haven't pulled the code down and tested it thoroughly, but glancing through it looks good. I'll follow up soon after I have played with it a bit.

warmwaffles committed:
Pass an array of objects to Report

Took the advice of @nateklaiber and made the Report less aware of what
defines 'HEALTHY' and passed in an Array of Services and Details.

warmwaffles committed:
Added options to be passed into a Detail

Updated README documentation

warmwaffles commented:
I will be merging this soon. Currently writing up an annotation for the `v1.0.0` tag.
